firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -53,6 +54,27 @@ public void tearDown() {
     IntegrationTestUtil.tearDown();
   }
 
+  /**
+   * Checks that running the query while online (against the backend/emulator) results in the same
+   * documents as running the query while offline. If `expectedDocs` is provided, it also checks
+   * that both online and offline query result is equal to the expected documents.
+   *
+   * @param query The query to check
+   * @param expectedDocs Ordered list of document keys that are expected to match the query
+   */
+  public void checkOnlineAndOfflineResultsMatch(Query query, String... expectedDocs) {
+    QuerySnapshot docsOnline = waitFor(query.get());
```

Comment:
Is it easier to do get(cache) and get(server)?

Reply:
why didn't I think of that? 😆  Done